### ISSUE TITLE
cake extension : java.lang.Throwable: Recursion prevented when computing FileIndex extension

### ISSUE THREAD

**Issue by abiezerm** [cake-build/cake-rider#160]
Plugin version: `1.0.0`
IDE version: `2021.1.3 (RD-211.7442.29)`
Operating system: `Mac OS X`
Exception: `java.lang.Throwable: Recursion prevented when computing FileIndex extension`
Stacktrace:
```text


StackTrace too long. Please copy the StackTrace here manually.


```

**Comment by nils-a** [comment]
Possible duplicate of #151 

@abiezerm Thanks for the report, can I ask you what you were doing when you got the error?
Is the error reproducible?

**Comment by abiezerm** [comment]
hey, @nils-a I just started the IDE, and I saw the error.

**Comment by nils-a** [comment]
ok. Thanks. Did it happen only once or is the error reproducible, somehow?

**Comment by abiezerm** [comment]
every time that I open the IDE

**Comment by nils-a** [comment]
So sorry to hear that. Could you add the StackTrace here?

**Comment by nils-a** [comment]
@abiezerm gentle ping - are you in a position to add a StackTrace here?

**Comment by abiezerm** [comment]
@nils-a  sorry for the delay, there you go:

java.lang.Throwable: Recursion prevented when computing FileIndex extension
	at com.intellij.javascript.nodejs.library.NodeModulesDirectoryManager.computeFileIndexExtensionPreventingRecursion(NodeModulesDirectoryManager.java:203)
	at com.intellij.javascript.nodejs.library.NodeModulesSyntheticLibraryProvider.getAdditionalProjectLibraries(NodeModulesSyntheticLibraryProvider.java:18)
	at com.intellij.openapi.roots.impl.RootIndex.buildRootInfo(RootIndex.java:243)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:72)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:56)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getRootIndex(DirectoryIndexImpl.java:171)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getRootIndex(DirectoryIndexImpl.java:154)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:181)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:98)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getInfoForFileOrDirectory(ProjectFileIndexImpl.java:27)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.isInContent(ProjectFileIndexImpl.java:196)
	at com.intellij.openapi.project.ProjectLocatorImpl.isUnder(ProjectLocatorImpl.java:61)
	at com.intellij.openapi.project.ProjectLocatorImpl.lambda$getProjectsForFile$1(ProjectLocatorImpl.java:75)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:844)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.intellij.openapi.project.ProjectLocatorImpl.getProjectsForFile(ProjectLocatorImpl.java:73)
	at net.cakebuild.language.CakeFileType.isMyFileType(CakeFileType.kt:23)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getByFile(FileTypeManagerImpl.java:648)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getFileTypeByFile(FileTypeManagerImpl.java:606)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getFileTypeByFile(FileTypeManagerImpl.java:595)
	at com.intellij.openapi.vfs.VirtualFile.getFileType(VirtualFile.java:354)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.getFileType(VirtualFileSystemEntry.java:578)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.computeCharset(VirtualFileSystemEntry.java:457)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.getCharset(VirtualFileSystemEntry.java:445)
	at com.intellij.openapi.vfs.VfsUtilCore.loadText(VfsUtilCore.java:376)
	at com.intellij.openapi.vfs.VfsUtilCore.loadText(VfsUtilCore.java:372)
	at com.intellij.javascript.nodejs.PackageJsonData.loadText(PackageJsonData.java:436)
	at com.intellij.javascript.nodejs.PackageJsonData.doParse(PackageJsonData.java:418)
	at com.intellij.javascript.nodejs.PackageJsonData.lambda$getOrCreate$16(PackageJsonData.java:402)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:844)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.javascript.nodejs.PackageJsonData.getOrCreate(PackageJsonData.java:392)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory$PackageJsonFiles.lambda$new$0(NodeModulesLibraryDirectory.java:505)
	at com.intellij.openapi.project.ProjectLocator.computeWithPreferredProject(ProjectLocator.java:61)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory$PackageJsonFiles.lambda$new$1(NodeModulesLibraryDirectory.java:504)
	at com.intellij.util.containers.ContainerUtil.map(ContainerUtil.java:1894)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory$PackageJsonFiles.<init>(NodeModulesLibraryDirectory.java:503)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory.doCreateLibrary(NodeModulesLibraryDirectory.java:117)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory.createLibrary(NodeModulesLibraryDirectory.java:105)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory.getLibrary(NodeModulesLibraryDirectory.java:93)
	at com.intellij.javascript.nodejs.library.NodeModulesSyntheticLibraryProvider.lambda$computeRoots$2(NodeModulesSyntheticLibraryProvider.java:29)
	at com.intellij.util.containers.ContainerUtil.mapNotNull(ContainerUtil.java:1977)
	at com.intellij.javascript.nodejs.library.NodeModulesSyntheticLibraryProvider.computeRoots(NodeModulesSyntheticLibraryProvider.java:25)
	at com.intellij.javascript.nodejs.library.NodeModulesSyntheticLibraryProvider.lambda$getAdditionalProjectLibraries$0(NodeModulesSyntheticLibraryProvider.java:18)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:43)
	at com.intellij.javascript.nodejs.library.NodeModulesDirectoryManager.computeFileIndexExtensionPreventingRecursion(NodeModulesDirectoryManager.java:197)
	at com.intellij.javascript.nodejs.library.NodeModulesSyntheticLibraryProvider.getAdditionalProjectLibraries(NodeModulesSyntheticLibraryProvider.java:18)
	at com.intellij.openapi.roots.impl.RootIndex.buildRootInfo(RootIndex.java:243)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:72)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:56)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getRootIndex(DirectoryIndexImpl.java:171)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getRootIndex(DirectoryIndexImpl.java:154)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:181)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:98)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getInfoForFileOrDirectory(ProjectFileIndexImpl.java:27)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.isInContent(ProjectFileIndexImpl.java:196)
	at com.intellij.openapi.project.ProjectLocatorImpl.isUnder(ProjectLocatorImpl.java:61)
	at com.intellij.openapi.project.ProjectLocatorImpl.lambda$getProjectsForFile$1(ProjectLocatorImpl.java:75)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:844)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.intellij.openapi.project.ProjectLocatorImpl.getProjectsForFile(ProjectLocatorImpl.java:73)
	at net.cakebuild.language.CakeFileType.isMyFileType(CakeFileType.kt:23)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getByFile(FileTypeManagerImpl.java:648)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getFileTypeByFile(FileTypeManagerImpl.java:606)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getFileTypeByFile(FileTypeManagerImpl.java:595)
	at com.intellij.openapi.vfs.VirtualFile.getFileType(VirtualFile.java:354)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.getFileType(VirtualFileSystemEntry.java:578)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.computeCharset(VirtualFileSystemEntry.java:457)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.getCharset(VirtualFileSystemEntry.java:445)
	at com.intellij.openapi.vfs.VfsUtilCore.loadText(VfsUtilCore.java:376)
	at com.intellij.openapi.vfs.VfsUtilCore.loadText(VfsUtilCore.java:372)
	at com.intellij.javascript.nodejs.PackageJsonData.loadText(PackageJsonData.java:436)
	at com.intellij.javascript.nodejs.PackageJsonData.doParse(PackageJsonData.java:418)
	at com.intellij.javascript.nodejs.PackageJsonData.lambda$getOrCreate$16(PackageJsonData.java:402)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:844)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.javascript.nodejs.PackageJsonData.getOrCreate(PackageJsonData.java:392)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory$PackageJsonFiles.lambda$new$0(NodeModulesLibraryDirectory.java:505)
	at com.intellij.openapi.project.ProjectLocator.computeWithPreferredProject(ProjectLocator.java:61)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory$PackageJsonFiles.lambda$new$1(NodeModulesLibraryDirectory.java:504)
	at com.intellij.util.containers.ContainerUtil.map(ContainerUtil.java:1894)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory$PackageJsonFiles.<init>(NodeModulesLibraryDirectory.java:503)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory.doCreateLibrary(NodeModulesLibraryDirectory.java:117)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory.createLibrary(NodeModulesLibraryDirectory.java:105)
	at com.intellij.javascript.nodejs.library.NodeModulesLibraryDirectory.getLibrary(NodeModulesLibraryDirectory.java:93)
	at com.intellij.javascript.nodejs.library.NodeModulesPredefinedLibraryProvider$NodeModulesLibraryModel.getSyntheticLibrary(NodeModulesPredefinedLibraryProvider.java:99)
	at com.intellij.lang.javascript.dialects.JSLanguageLevelPusher.addFilesFromModel(JSLanguageLevelPusher.java:247)
	at com.intellij.lang.javascript.dialects.JSLanguageLevelPusher.getApplicableLibraryFiles(JSLanguageLevelPusher.java:235)
	at com.intellij.lang.javascript.dialects.JSLanguageLevelPusher.lambda$doUpdateJSLibraries$3(JSLanguageLevelPusher.java:216)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:848)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.lang.javascript.dialects.JSLanguageLevelPusher.doUpdateJSLibraries(JSLanguageLevelPusher.java:211)
	at com.intellij.lang.javascript.dialects.JSLanguageLevelPusher.lambda$updateJSLibraries$0(JSLanguageLevelPusher.java:205)
	at com.intellij.util.ui.update.Update$1.run(Update.java:108)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:333)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:323)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:273)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:287)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:242)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:369)
	at com.intellij.util.Alarm$Request.run(Alarm.java:355)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:220)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)

**Comment by nils-a** [comment]
@abiezerm Thank you. May I ask what kind of project the underlying project is?

**Comment by nils-a** [comment]
I was able to reproduce the error. It occurs only in RD- versions 211.7442.x or newer, not in earlier versions and only when projects without solutions (e.g. node-projects) are involved.

**Comment by nils-a** [comment]
@abiezerm , @aroige, @manuc66, @Slampen 
If anyone wanted to test an unstable development-build: This version should have this bug fixed: https://ci.appveyor.com/project/cakebuild/cake-rider/builds/40011544/artifacts

Be aware that there are some breaking changes in the upcoming [version 2.0.0](https://github.com/cake-build/cake-rider/milestone/6?closed=1)
